hazelcast/src/main/java/com/hazelcast/jet/core/processor/SourceProcessors.java

```diff
@@ -406,19 +413,21 @@ public static ProcessorMetaSupplier streamFilesP(
      */
     @Nonnull
     public static <T> ProcessorMetaSupplier streamJmsTopicP(
+            @Nullable String destination,
+            boolean isSharedConsumer,
+            @Nonnull ProcessingGuarantee maxGuarantee,
+            @Nonnull EventTimePolicy<? super T> eventTimePolicy,
             @Nonnull SupplierEx<? extends Connection> newConnectionFn,
             @Nonnull FunctionEx<? super Session, ? extends MessageConsumer> consumerFn,
-            boolean isSharedConsumer,
             @Nonnull FunctionEx<? super Message, ?> messageIdFn,
-            @Nonnull FunctionEx<? super Message, ? extends T> projectionFn,
-            @Nonnull EventTimePolicy<? super T> eventTimePolicy,
-            ProcessingGuarantee maxGuarantee
+            @Nonnull FunctionEx<? super Message, ? extends T> projectionFn
     ) {
         ProcessorSupplier pSupplier = new StreamJmsP.Supplier<>(
-                newConnectionFn, consumerFn, messageIdFn, projectionFn, eventTimePolicy, maxGuarantee);
+                maxGuarantee, eventTimePolicy, newConnectionFn, consumerFn, messageIdFn, projectionFn);
+        ConnectorPermission permission = ConnectorPermission.jms(destination, ACTION_READ);
         return isSharedConsumer
-                ? ProcessorMetaSupplier.of(StreamJmsP.PREFERRED_LOCAL_PARALLELISM, pSupplier)
```

Comment:
`StreamJmsP.PREFERRED_LOCAL_PARALLELISM` has no usage anymore, we can remove it.